src/cloudwatch.ts

```diff
@@ -1,30 +1,27 @@
-import { ClientConfiguration as cwConfig, PutMetricDataInput } from "aws-sdk/clients/cloudwatch";
-import { CloudWatch } from "aws-sdk";
+import CloudWatch, { ClientConfiguration as cwConfig, PutMetricDataInput } from "aws-sdk/clients/cloudwatch";
 import { DateTime } from "luxon";
 import { cwLogger } from "./handler";
 import { CloudWatchLogsLogEvent } from "aws-lambda";
-import AWSXRay from "aws-xray-sdk";
+import AWSXRay, { Subsegment } from "aws-xray-sdk";
 import RE2 = require("re2");
 
 export class CW {
-  private readonly config: cwConfig;
+  private readonly config: cwConfig = { region: "eu-west-1", retryDelayOptions: { base: 500 } };
   private readonly branch: string = (process.env.BRANCH ?? "local").toLocaleLowerCase();
   private readonly now: DateTime = DateTime.utc();
 
-  public constructor(cwConf?: cwConfig) {
-    this.config = cwConf ?? { region: process.env.AWS_REGION ?? "eu-west-1", retryDelayOptions: { base: 500 } };
-  }
-
   /**
    * Pushes the visit metrics to CloudWatch.
+   *
    * @public
    * @async
-   * @param visitsToday Number of visits since 00:00 UTC today.
-   * @param oldVisits Number of visits since now UTC-10 hours.
-   * @param openVisits Amount of open visits right now.
-   * @returns {string} Combination of visit metrics for response.
+   * @param {number} visitsToday Number of visits since 00:00 UTC today.
+   * @param {number} oldVisits Number of visits since now UTC-10 hours.
+   * @param {number} openVisits Amount of open visits right now.
+   * @param {Subsegment} parentSubSeg The parent AWS X-Ray subsegment
    */
-  public async sendVisits(visitsToday: number, oldVisits: number, openVisits: number): Promise<string> {
+  public async sendVisits(visitsToday: number, oldVisits: number, openVisits: number, parentSubSeg?: Subsegment): Promise<void> {
```

Comment:
For this return I wonder whether we might want to return a type so it helps us debug the code in the future otherwise we may have to check for `instance of`/`typeof` things or make sure we don't have undefined returned. Perhaps we need a broader discussion ?

Reply:
This used to return a string which would then only be logged, since the logging is done in this method anyway there is no point to return a value.

Reply:
I understand we are consuming it for now but i think it is better if we can return the timeoutCount from the function as it seems it what changes depending on the logEvents (and this is what we publish to amazon cloudwatch. This will make the unit testing better as we are likely to mock that and if for whatever reasons this was not mocked properly we would assert on undefined and will make it hard to to know whether the unit test passes because the function did its job or because the mock is undefined (or any other weird undefined behaviour happening with js)